pkg/util/kvgraph/util.go

```diff
@@ -68,25 +68,57 @@ func addCommonVMIObjectGraph(spec v1.VirtualMachineInstanceSpec, vmName, namespa
 }
 
 func addVolumeGraph(vmiSpec v1.VirtualMachineInstanceSpec, vmName, namespace string, resources []velero.ResourceIdentifier) ([]velero.ResourceIdentifier, error) {
+	seen := make(map[string]struct{}) // Deduplication map
+
+	// Local wrapper: We need this since UserDataSecretRef and NetworkDataSecretRef may contain the same reference and we don;t want duplicates.
```

Comment:
We don't usually worry about duplicates since velero discards them if an object were backed up twice. Is there any reason for this?

Reply:
Yep as @alromeros mentioned you can add the same object to the list and velero handles that for you.
So only need to add the new cases. Thanks

Reply:
Within the internal velero item actions, there are cases which will always produce duplicates (returning VolumeSnapshotClass for each VolumeSnapshot, for example) -- this is fine. Velero maintains a map of items already handled, so it will skip items already backed up.

Reply:
No particular reason for removing duplicates apart from having to return unique ones. I'll remove the code for de-dupe and commit the change for review. Thank you.

Reply:
https://github.com/kubevirt/kubevirt-velero-plugin/pull/359/commits/2eced8dcffdc4b46e11b8bdd69e3abb3dc35ce1c

Implemented the suggestions in review comments.